icechunk/src/storage/mod.rs

```diff
@@ -257,9 +257,44 @@ pub async fn make_storage(
             )
             .await?,
         ),
+        ObjectStoreConfig::Gcs(options) => {
+            let mut options = options.unwrap_or_default();
 
-        #[allow(clippy::unimplemented)]
-        ObjectStoreConfig::Gcs {} => unimplemented!(),
+            #[allow(clippy::expect_used)]
+            match credentials {
+                Some(Credentials::Gcs(GcsCredentials::ServiceAccount(path))) => {
+                    options.push((
+                        "google_service_account".to_string(),
+                        path.into_os_string()
+                            .into_string()
+                            .expect("Invalid path to google service account credentials"),
+                    ));
+                }
+                Some(Credentials::Gcs(GcsCredentials::ServiceAccountKey(key))) => {
+                    options.push(("google_service_account_key".to_string(), key));
+                }
+                Some(Credentials::Gcs(GcsCredentials::ApplicationCredentials(path))) => {
+                    options.push((
+                        "google_application_credentials".to_string(),
+                        path.into_os_string()
+                            .into_string()
+                            .expect("Invalid path to google application credentials"),
+                    ))
+                }
+                _ => {}
```

Comment:
let's make this `None` instead of `_` so we get an error if we ever add anything to the enum